packages/frontend/src/hooks/useTableConfig.ts

```diff
@@ -84,7 +101,131 @@ const useTableConfig = (
 
     const totals = useColumnTotals({ resultsData, itemsMap });
 
+    const pivotDimension = pivotDimensions?.[0];
+    const uniquePivotValues: Record<string, { raw: any; formatted: any }> =
+        useMemo(
+            () =>
+                pivotDimension && itemsMap[pivotDimension] && resultsData
+                    ? resultsData.rows.reduce<
+                          Record<string, { raw: any; formatted: any }>
+                      >((acc, row) => {
+                          const value = row[pivotDimension].value;
+                          return { ...acc, [`${value.raw}`]: value };
+                      }, {})
+                    : {},
+            [pivotDimension, itemsMap, resultsData],
+        );
+    const exceedsMaxPivotValues = useMemo<boolean>(
+        () => Object.keys(uniquePivotValues).length > 20,
+        [uniquePivotValues],
+    );
+
     const columns = useMemo(() => {
+        if (exceedsMaxPivotValues) {
+            return [];
+        }
+
+        if (
```

Comment:
Just to check I understand, the headers are being built separately from the data? which happens in the getPlotData function?

Are we sure they will always be in the same order? Or is the order not important from the plot data?

It would be great to centralise the pivoting behaviour so it goes:

1. First take raw data + pivot configuration
2. Produce the pivot table, including all the column information
3. Everything downstream (like table component) just needs the new columns and rows produced in step 2 above